cpp/tests/io/parquet_chunked_reader_test.cu

```diff
@@ -953,64 +953,283 @@ TEST_F(ParquetChunkedReaderTest, TestChunkedReadNullCount)
-                                              cudf::table_view{{expected}})
-      // note: it is unnecessary to force compression to NONE here because the size we are using in
-      // the row group is the uncompressed data size. But forcing the dictionary policy to
-      // dictionary_policy::NEVER is necessary to prevent changes in the
-      // decompressed-but-not-yet-decoded data.
-      .dictionary_policy(cudf::io::dictionary_policy::NEVER);
-
-  cudf::io::write_parquet(opts);
-
-  {
-    // no chunking
-    auto const [result, num_chunks] = chunked_read(filepath, 0, 0);
-    EXPECT_EQ(num_chunks, 1);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->get_column(0));
-  }
+  cudf::io::parquet_writer_options out_opts =
+    cudf::io::parquet_writer_options::builder(cudf::io::sink_info{filepath}, t)
+      .compression(compression)
+      .dictionary_policy(dict_policy);
+  cudf::io::write_parquet(out_opts);
+}
 
-  {
-    // 25 chunks of 100k rows each
-    auto const [result, num_chunks] = chunked_read(filepath, 0, 1);
-    EXPECT_EQ(num_chunks, 25);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->get_column(0));
-  }
+void input_limit_test_write(std::string const& base_path, cudf::table_view const& t)
+{
+  // no compression
+  input_limit_test_write_one(base_path + "_a.parquet",
+                             t,
+                             cudf::io::compression_type::NONE,
+                             cudf::io::dictionary_policy::NEVER);
+  // compression with a codec that uses a lot of scratch space at decode time (2.5x the total
+  // decompressed buffer size)
+  input_limit_test_write_one(base_path + "_b.parquet",
+                             t,
+                             cudf::io::compression_type::ZSTD,
+                             cudf::io::dictionary_policy::NEVER);
+  // compression with a codec that uses no scratch space at decode time
+  input_limit_test_write_one(base_path + "_c.parquet",
+                             t,
+                             cudf::io::compression_type::SNAPPY,
+                             cudf::io::dictionary_policy::NEVER);
+  input_limit_test_write_one(base_path + "_d.parquet",
+                             t,
+                             cudf::io::compression_type::SNAPPY,
+                             cudf::io::dictionary_policy::ALWAYS);
+}
 
-  {
-    // 25 chunks of 100k rows each
-    auto const [result, num_chunks] = chunked_read(filepath, 0, 4000000);
-    EXPECT_EQ(num_chunks, 25);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->get_column(0));
-  }
+void input_limit_test_read(std::string const& base_path,
+                           cudf::table_view const& t,
+                           size_t output_limit,
+                           size_t input_limit,
+                           int const expected_chunk_counts[4])
+{
+  std::vector<std::string> file_suffix{"_a.parquet", "_b.parquet", "_c.parquet", "_d.parquet"};
```

Comment:
I feel like it would be simpler to generate the full file names in a helper function and pass the vector of names to input_limit_test_read/input_limit_test_write, rather then assembling the names in multiple places.